docs/website/contents/for-developers/HardForkCombinatorRollbackMonotonicityProposal.md

```diff
@@ -0,0 +1,285 @@
+
+So many dimensions arose from investiating such a relatively simple misbehavior for two reasons.
+First, the mechanisms implementing these policies are---at least in the current implementation---somewhat interdependent.
+Second, the HFC (including its instantiation for Cardano) remains underdocumented since back when it was originaly developed under time-pressure, and our team no longer includes its inventor.
+The Consensus Team learns new things and unfortunately re-learns some old things every time they dive in to alter it.
+(Optimistically, Nicolas Frisby feels they currently understand it now better than they have ever before.)
+
+## Dimensions
+
+The following discrete variables characterize dimensions of the HFC design space.
+In this context, the _determination_ is the `singleEraTransition` method: it's how an era's ledger state determines when the era ends.
+
+- DST - Determination is Sensitive to Ticks  - whether ticking the ledger state can change its determination of the end of its era: Sensitive vs Insensitive
+
+- CCT - Criss-Crossed Ticks - whether an era's tick rules might be used to tick across slots that are in _other_ eras: Allowed vs Disallowed
+
+- PIL - Potential to Ignore the Ledger - whether the HFC can ignore the ledger's hard fork-related governance: Possible vs Impossible
+
+- DSR - Determination is Sensitive to Rollback - for chains C and D where C < D and len(C) <= len(C /\ D) + k, whether the current era of C can have a determined end on C but an undetermined or different end on D: Sensitive vs Insensitive
+
+- ZEE - Zero Epoch Eras - whether an era (in its entirety) can contain zero epochs: Allowed vs Disallowed
+
+- ZBE - Zero Block Eras - whether an era (in its entirety) can contain zero blocks: Allowed vs Disallowed
+
+- QSR - Queries are Sensitivity to Rollback - same as DSR but for HFC-related Node-To-Client queries instead of for validity of blocks: Sensitive vs Insensitive
+
+Note the following relations.
+
+- ZEE=Allowed implies ZBE=Allowed.
+- DSR=Insensitive implies QSR=Insensitive.
+
+## Table for the Current Cardano HFC
+
+The following table and associated notes classifies the current HFC implementation in terms of these dimensions.
+
+| DST | CCT | PIL | DSR | ZEE | ZBE | QSR |
+| --- | --- | --- | --- | --- | --- | --- |
+| I   | A   | P 1 | I   | 2   | 3   | I   |
+
+- (1) It's Possible, but only if there's a Chain Growth violation at a particular time during the epoch.
+
+- (2) The current HFC's behavior is inconsistent without making further assumptions.
+  Overall, ZEE is allowed, but it might cause some unexpected behavior.
+  In particular, forecasts always anticipate that there is at least one safe zone of slots in an era (which is not true for eras with zero epochs).
+  That mismatch could manifest as incorrect time-slot translations eg (within ledger rules execution, but not within Node-To-Client query reponses).
```

Comment:
Could we elaborate an example of this?

Reply:
We could. It should be pretty simple, but it will require a lot of words, just because of the context. If it doesn't require a lot of works, then it would remain pretty abstract, which kind of defeats the purpose of the example.

I left it out for now because it seems like a lot of words. Maybe we could consolidate all of the examples into an "example" section that shares the necessary context amongst a few pointed examples.

Reply:
Yeah, I think that would help the readers greatly. But I acknowledge this is not intended to be a document used for mass consumption, right? So perhaps this is also out of scope.

Reply:
Yeah, I think that would help the readers greatly. But I acknowledge this is not intended to be a document used for mass consumption, right? So perhaps this is also out of scope.